src/NServiceBus.MongoDB/SagaPersister/MongoSagaPersister.cs

```diff
@@ -177,8 +177,8 @@ private static void CheckUniqueProperty(IContainSagaData sagaData, KeyValuePair<
             Contract.Requires(!string.IsNullOrWhiteSpace(property));
             Contract.Requires(value != null);
 
-            var query = Query.EQ(property, value.ToString());
-
+            var query = (value is Guid) ? Query.EQ(property, (Guid)value) : Query.EQ(property, value.ToString());
```

Comment:
Yeah. We need to release a patch for one app which cannot migrate to nsb 5.x as of now...


> On Apr 14, 2015, at 2:48 PM, Wes Day <notifications@github.com> wrote:
> 
> In src/NServiceBus.MongoDB/SagaPersister/MongoSagaPersister.cs:
> 
> > @@ -177,8 +177,8 @@ private static void CheckUniqueProperty(IContainSagaData sagaData, KeyValuePair<
> >              Contract.Requires(!string.IsNullOrWhiteSpace(property));
> >              Contract.Requires(value != null);
> >  
> > -            var query = Query.EQ(property, value.ToString());
> > -
> > +            var query = (value is Guid) ? Query.EQ(property, (Guid)value) : Query.EQ(property, value.ToString());
> Looks like this was already fixed (though it requires upgrading to NServiceBus 5).
> 
> —
> Reply to this email directly or view it on GitHub.
>